### PR TITLE
implement greasing of the QUIC bit (RFC 9287)

### LINKED ISSUE #3133: implement the QUIC Grease Bit (RFC 9287)
https://www.rfc-editor.org/rfc/rfc9287.html

As the QUIC bit can be used for demultiplexing QUIC with other protocols, we should probably add a `quic.Config` option to turn this off: `DisableQUICBitGreasing`.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #3133.

We don't (yet) save the value of this transport parameter. This means that we'll never grease the QUIC bit on the first Initial packet and on 0-RTT packets. Implementing this is left as an exercise for a later PR.

### CODE REVIEW

**Review by lucas-clemente — APPROVED** [review]

**Comment by marten-seemann** [comment]
This PR is on hold until we've merged all the header changes planned for #3526.

**Comment by marten-seemann** [comment]
Closing this PR, since we won't implement RFC 9287 in quic-go. See https://github.com/quic-go/quic-go/issues/3133#issuecomment-1862118677 for reasoning.

**Pull request closed without merging**
